src/App.js

```diff
@@ -16,50 +19,67 @@ class App extends React.Component {
     fetch('https://api.punkapi.com/v2/beers')
       .then(data => data.json())
       .then(beers => {
-        console.log('beers', beers)
-        this.setState({beers: beers});
+        console.log('beers', beers);
+        this.setState({ beers });
       })
   }
 
-  checkLiked = (beer) => {
-    if (this.state.likedBeers.find(item => item.id === beer.id)) {
-      console.log('found, deleting')
-      this.setState(prevState => {
-        return {
-          likedBeers: [...prevState.likedBeers.filter(item => item.id !== beer.id)]
-        }
-      })
+  toggleLiked = (beer) => {
+    const { id: beerId } = beer;
+    const { likedBeers } = this.state;
+
+    if (likedBeers.find(({id}) => id === beerId)) {
+      console.log('found, deleting');
+      this.setState(({ likedBeers }) => ({
+        likedBeers: likedBeers.filter(({ id }) => id !== beerId),
+      }));
     } else {
-      console.log('adding...')
-      this.setState(prevState => {
-        return {
-          likedBeers: [...prevState.likedBeers, beer]
-        }
-      })
+      console.log('adding...');
+      this.setState(({ likedBeers }) => ({
+        likedBeers: [...likedBeers, beer],
+      }));
     }
-    // log # of likedBeers in state, delay for async
-    setTimeout(() => console.log('likedBeers count: ', this.state.likedBeers.length), 300)
+  };
 
-  }
+  home = () => {
+    const { beers, likedBeers } = this.state;
 
-  home = () => <Home checkLiked={(beer) => this.checkLiked(beer)} beers={this.state.beers}/>;
-  myList = () => <MyList likedBeers={this.state.likedBeers}/>;
+    return (
+      <Home
+        toggleLiked={this.toggleLiked}
+        beers={beers}
+        likedBeers={likedBeers}
+      />
+    );
+  };
+
+  pageNotFound = () => (
+    <h1>
+      Page not found
+    </h1>
+  );
+
+  myList = () => (
+    <MyList
+      toggleLiked={this.toggleLiked}
+      likedBeers={this.state.likedBeers}
+    />
+  );
 
   render() {
+    console.log('likedBeers count: ', this.state.likedBeers.length);
```

Comment:
Moved from setTimeout (BAD) into render. If you truly wanted this log to only fire when specifically when `state.likedBeers` changes, you could:
```
componentDidUpdate(prevProps, ({ likedBeers: prevLikedBeers })) {
  const { likedBeers } = this.state;
  if (likedBeers.length !== prevLikedBeers.length) console.log(`likedBeers count: ${likedBeers.length}`);
}
```